### PR TITLE
feat: add dotty candidate

### PR DESCRIPTION
**Pull request by mingchuno**

### CODE REVIEW

File: src/main/scala/io/sdkman/changelogs/DottyMigrations.scala [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+  @ChangeSet(
+    order = "001",
+    id = "001_add_dotty_candidate",
+    author = "mingchuno"
+  )
+  def migration001(implicit db: MongoDatabase): Candidate = {
+    Candidate(
+      candidate = "dotty",
+      name = "Dotty",
+      description = "Research compiler that will become Scala 3",
```
Comment (marc0der) [inline src/main/scala/io/sdkman/changelogs/DottyMigrations.scala]: Can we add a bit more detail here please? This is what gets displayed on our website at https://sdkman.io/sdks

**Comment by mingchuno** [comment]
@marc0der Done adding some more context on the desc. Please check

**Comment by marc0der** [comment]
Thank you!

**Comment by marc0der** [comment]
@mingchuno Sorry, but the [build is red](https://travis-ci.org/github/sdkman/sdkman-db-migrations/builds/667195588) now. Would you mind taking a look at what the problem is please?

**Comment by mingchuno** [comment]
@marc0der I have just check. The `0.21.0` version is a trap! Its file name is `0.21.0-RC1`
https://github.com/lampepfl/dotty/releases/tag/0.21.0
The bundle size is larger than usual too...
Let me made another PR to remove it, okay?

**Comment by marc0der** [comment]
Thanks for looking into it. Please do try running these migrations locally before submitting a PR in future. This will help us catching such problems earlier.

**Pull request merged**